README.md
![Logo Albulle](http://doc.jebulle.net/_media/apps/albulle1.0/logo_albulle.png "Albulle, galerie photos")

http://albulle.jebulle.net • http://demos.jebulle.net/albulle/

# Albulle

**Albulle** est un logiciel **PHP** de publication de **galeries photos** sans base de données (c'est sa raison de vivre initiale et pour l'instant, ça l'est toujours). Démarré en 2005 et activement maintenu jusqu'en 2010, la vie a fait son oeuvre et j'ai arrêté de maintenir le projet. Son objectif reste de proposer un outil simple d'utilisation pour **mettre en ligne vos photos**. 

10 ans plus tard, le mail d'un utilisateur d'Albulle m'a relancé sur le projet. Et c'est ainsi que la version 1.3 voit le jour ! Elle rend compatible Albulle avec **PHP7** en plus de proposer un thème moins "sucré". Changement des icônes, simplification des feuilles de styles, passage à la norme **HTML5**, utilisation de **FancyBox** à la place de **LightBox**, rien de révolutionnaire mais qui permet à Albulle de continuer à faire son job. Cette version 1.3 ne propose pas encore d'interface **responsive**.

Par ailleurs, j'ai profité de cette version 1.3 pour publier le projet sur Github. Vous pourrez y retrouver l'ensemble des anciennes releases et désormais, les nouvelles évolutions.

## Licence

Albulle est distribué sous la licence CeCILL2. Les détails de celle-ci sont disponibles dans le fichier correspondant. L'utilisation d'Albulle équivaut à une acceptation tacite de sa licence.

## Installation

La procédure d'installation est disponible à l'adresse suivante : http://doc.jebulle.net/apps/albulle1.0/

## Crédits

- [FancyBox](https://www.fancyapps.com/fancybox/3/) qui remplace LightBox depuis la version 1.3,
- [Fugue Icons](https://p.yusukekamiyamane.com/) qui remplacent les anciennes icônes XP iCandy.

## Anciennes notes

### Edition des fichiers

Le projet est écrit en **UTF-8** (sans BOM). De nos jours, tous les éditeurs modernes proposent un fonctionnement compatible avec cette norme. Assurez-vous d'être dans la bonne configuration avant d'éditer les fichier (mais la détection automatique est désormais de mise dans tous les éditeurs).

### Installation chez Free.fr

En 2010, l'installation chez Free posait quelques problèmes. Ne sachant si ces problèmes sont toujours d'actualité, je conserve pour l'instant les notes qui s'appliquaient à l'époque (dispos dans le répertoire `extras/free.fr/`). Elles seront peut-être amené à disparaître.

## Modification des fichiers de thème `*.thm.php``

L'avertissement joint à tous les lisezmoi.txt (que je fais disparaître au profit du désormais standard README.md) reste encore valable en 2020 : assurez-vous que les fins de lignes de ces fichiers soient bien de type UNIX (= LF).

changelog.txt
:: Albulle
:: Version 1.4

WHAT's NEW :
------------

Version 1.4 - 21/09/2022

	* Ajouté : thème albulle responsive
	* Modifié : bouton "Mode galerie/diaporama" remplacé par des icônes
	* Modifié : nouvelle présentation des vignettes (carrées et plus grandes)
	* Modifié : déplacement de la limite des dimensions des images dans une constante (JB_AL_PHOTOS_MAX_PIXELS)

Version 1.3 - 30/05/2020

	* Ajouté : FancyBox remplace LightBox,
	* Modifié : passage à l'HTML5 pour la structure du thème,
	* Modifié : fusion des fichiers CSS dans un seul fichier,
	* Modifié : retrait de l'affichage de la version d'Albulle,
	* Modifié : rafraichissement visuel du thème par défaut,
	* Modifié : mise à jour de jQuery,
	* Modifié : augmentation de la qualité par défaut des miniatures (80 > 90),
	* Modifié : retrait de l'usage des "id" depuis CSS (au profit de "class"),
	* Modifié : retrait de Cooliris (mort depuis longtemps !),
	* Corrigé : compatibilité avec PHP7.

Version 1.2.1 - 25/05/2010 - rev. 38 > 39

	* Ajouté : fourniture du nécessaire pour l'installation chez Free.fr dans les extras,
	* Modifié : configuration du fuseau horaire faite depuis jblib,
	* Corrigé : rendu en mode intégration à un site. $sThmCssMetas devient $sAlbulleMetas.

Version 1.2 - 07/05/2010 - rev. 23 > 36

	* Ajouté (FS#9)  : refonte du moteur de rendu HTML et par conséquent du thème par défaut,
	* Ajouté (FS#10) : tri des photos selon la date de prise de vue,
	* Ajouté (FS#3)  : lecture de la légende IPTC des images (nouvelle pseudo variable {$legende}),
	* Modifié (FS#11) : sortie du code XHTML de la méthode afficherExif(),
	* Modifié : html.php et html_popup.php renommés en render.php et render_popup.php,
	* Modifié : légère modification du logo,
	* Corrigé (FS#5) : correction ajout des fichiers au panier depuis un lien symbolique.

Version 1.1.1 - 01/10/2009 - rev. 17 > 21

	* Ajouté : nouveau bloc conditionnel "SI ACCUEIL" dans les thèmes (FS#8)
	* Ajouté : le bloc conditionnel "SI MODE_GALERIE" est disponible dans tous les fichiers d'un thème (FS#8)
	* Corrigé : erreur encodage UTF-8 des noms de dossiers (FS#4)

Version 1.1 - 12/11/2008 - rev. > 15

	* Ajouté : flux RSS pour compatibilité avec Cooliris
	* Ajouté : possibilité de placer un fichier texte.html dans un dossier pour afficher un résumé dans la page du dossier
	* Ajouté : nouvelle constante JB_AL_ACCUEIL_ALT qui permet de surcharger le texte d'accueil
	* Modifié : renommage des classes CSS du menu de pagination .page -> .btn-page et .pageActive -> .btn-page-active
	* Modifié : LightBox2 remplacé par jQuery.Lightbox
	* Corrigé : mauvaise URL du lien vers la page d'accueil dans le cas d'une intégration
	* Corrigé : concaténation URL pour prendre en compte les sites qui utilisent la réécriture d'URL

Version 1.0 - 16/02/2008

	* Ajouté : création d'une miniature par défaut pour les images corrompues qui auraient réussi à passer
	  au travers des mailles du filet !
	* Ajouté : nouveau paramètre JB_AL_FICHIERS_UTF8 pour la gestion des accents. Cela permet d'indiquer
	  à Albulle que le système de fichier où il est hébergé est en UTF8
	* Modifié : lecture des images pour ne plus prendre les images corrompues
	* Corrigé : oubli de la déclaration globale de l'objet $oOutils dans la fonction cheminDansPanier()
	* Corrigé : vérification des fichiers du panier qui ne fonctionnait pas sous Windows et qui rendait
	  le panier inopérant (problème de séparateur des dossiers)
	* Corrigé : variables indéfinies en mode panier inactif
	* Corrigé : tous les problèmes liés aux accents. Enfin !

Version 1.0-rc6 - 25/06/2007

	* Correction erreur de création des miniatures avec les photos qui possèdent leur double dans
	  le dossier originales,
	* Correction d'une notice qui restait dûe à la présence résiduelle de l'accès à
	  $aActions['diaporama'] qui n'existe plus depuis la version 1.0-rc4.

Version 1.0-rc5 - 19/06/2007

	* Reconversion des fins de lignes de l'intégralité des fichiers en LF(Unix). Tout le monde était
	  repassé en CR+LF (Windows) ce qui posait problème pour la construction des pages depuis
	  les fichiers du thème,
	* Correction du fichier index.php qui était livré dans un mauvais encodage avec la 1.0-rc4,
	* Suppression des masques \t\t dans les expressions régulières de remplacement.

Version 1.0-rc4 - 08/06/2007

	* Ajout paramètre activation/désactivation du panier,
	* Ajout du paramétrage du mode d'affichage par défaut : galerie ou diaporama,
	* Ajout de la possibilité de stocker les photos selon deux qualités. Cela peut permettre
	  d'afficher des images de dimensions réduites tout en les téléchargeant dans leur qualité originale,
	* Modification de certaines images de fond en PNG pour alléger le thème par défaut,
	* Correction des masques '.*' des expressions régulières de reconnaissance des blocs conditionnels
	  en '.*?' pour obtenir des expressions régulières non gourmandes,
	* Correction d'une faille d'accès aux fichiers PHP depuis le panier.

Version 1.0-rc3 - 06/01/2007

	* Correction bogue des entêtes envoyées lors du téléchargement de l'archive pour les installations
	  sur serveur Windows.

Version 1.0-rc2 - 30/11/2006

	* Correction bogue du panier vide,
	* Corrections pour la gestion de la sortie iso-8859-1 pour l'intégration dans un site.

Version 1.0-rc1 - 15/11/2006

	* Correction de l'appel invalide du fichier blank.gif dans les css du thème par défaut,
	* Correction de l'affichage de l'icone ajout / retrait du panier sur les vignettes,
	* Correction erreur fonction inclure lors du téléchargement,
	* Correction sémantique XHTML du texte d'accueil,
	* Correction pixel d'écart sur le bloc détail de la boite lightbox,
	* Corrections class url.class.php (undefined index),
	* Amélioration gestion des vignettes,
	* Ajout de la gestion des accents pour les noms de dossiers et fichiers,

medias/html.php
<?php echo "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n"; ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://wwww.w3.org/1999/xhtml">
	<head>
		
		<link rel="stylesheet" href="./medias/style.css" type="text/css" />
		
		<title>.: AlBulles :. .: <?php echo $sHeadTitre; ?> :.</title>
 	
	</head>
	
	<body>
	
		<!-- DEBUT droite -->
		<div class="droite">
			
			<h1><?php echo $sBodyTitre; ?></h1>
			
			<?php
			if ( empty( $sRep ) )
			{
			?>
			
			<div class="accueil">
			
				<strong>Bienvenue dans AlBulles !</strong>
				
				<br /><br />
				
				AlBulles est un programme de galerie photos.<br /><br />
				
				Ce texte est un exemple. Vous pouvez le remplacer par ce que vous souhaitez. Pour consulter les photos
				disponibles, utilisez la liste des dossiers.<br /><br />
				
				Un panier virtuel est &agrave; votre disposition. Il vous suffit d'ajouter les images &agrave; votre panier pour
				ensuite les t&eacute;l&eacute;charger sous forme d'archive zip.
				
			</div>
			
			<?php
			}
			else {
			?>
			
			<!-- DEBUT barre de navigation -->
			<div class="navigation">
				<a href="<?php echo $_SERVER['PHP_SELF']; ?>" class="page">Accueil</a>
				Pages : <?php echo $sNavigation; ?>
			</div>
			<!-- FIN barre de navigation -->
			
			<?php
			for ( $i = 0 ; $i < sizeof( $aMiniatures ) ; $i++ ) { ?>
			<!-- DEBUT une miniature -->
			<div class="miniature">
			
				<?php echo $aMiniatures[$i]['LIEN_PHOTO']; ?>
				
				<div class="infosImg">
					<?php
					echo $aMiniatures[$i]['DIM_PHOTO'].'<br />';
					echo $aMiniatures[$i]['SIZE_PHOTO'];
					?>
				</div>
				
				<div class="puce"><?php echo $aMiniatures[$i]['AJOUT_PANIER']; ?></div>
				
			</div>
			<!-- FIN une miniature -->
			
			<?php
			} // for
			} // if
			?>
			
			<div class="spacer"></div>
		
		</div>
		<!-- FIN droite -->
	
		<!-- DEBUT gauche -->
		<div class="gauche">
		
			<!-- DEBUT liste dossiers photos -->
			<div class="dossiers">
			
				<img src="./medias/images/albulles_dossiers_dispos.jpg" alt="Dossiers des photos" />
				
				<ul class="liens">
					<?php echo $sLiensDossiersPhotos; ?>
				</ul>
				
				<div class="spacer"></div>
			
			</div>
			<!-- FIN liste dossiers photos -->
			
			<!-- DEBUT barre de gestion du panier -->
			<div class="panier">
			
				<img src="./medias/images/albulles_panier.jpg" alt="Panier" /><br /><br />
				
				Fichiers dans le panier : <br /><strong><?php echo $sNbFichiersDansLePanier; ?></strong>
				
				<br /><br />
				
				<div class="actions">
					<?php echo $sPanierLienArchive,"\n",$sPanierLienVider,"\n"; ?>
				</div>
				
				<br />
				
				<?php echo $sPanierLienToutAjouter,"\n",$sPanierLienToutSupprimer,"\n"; ?>
			
			</div>
			<!-- FIN barre de gestion du panier -->
			
			<!-- DEBUT Copyright -->
			<div class="copyright">
			
				<a href="http://www.mozilla.eu.org/fr/products/firefox/" title="Ce site s'affiche mieux avec un navigateur respectant les normes">
					<img src="./medias/images/firefox_80x15.png" width="80" height="15" title="Ce site s'affiche mieux avec un navigateur respectant les normes" alt="T&eacute;l&eacute;chargez FireFox" />
				</a>
				
				<br />
				
				<a href="http://jebulle.net/index.php?rubrique=albulles" title="T&eacute;l&eacute;chargez AlBulles">
					<img src="./medias/images/AlBulles_80x15.png" width="80" height="15" title="T&eacute;l&eacute;chargez AlBulles" alt="T&eacute;l&eacute;chargez AlBulles" />
				</a>
				
				<br />
				
				AlBulles <?php echo $sVersion; ?> &copy; <a href="http://jebulle.net">Bubulles Creations</a> - 2005
			
			</div>
			<!-- FIN Copyright -->
		
		</div>
		<!-- FIN gauche -->
	
	</body>
	
</html>